release.js

```diff
@@ -36,15 +36,30 @@ module.exports = async function ({ github, context, inputs }) {
 
   const { opticUrl, npmTag, version, id } = releaseMeta
 
+  const run = runSpawn()
   if (!pr.merged) {
-    return github.rest.repos.deleteRelease({
-      owner,
-      repo,
-      release_id: id,
-    })
+    const branchName = `release/${version}`
+    try {
+      await run('git', ['push', 'origin', '--delete', branchName])
+    } catch (err) {
+      return core.setFailed(
+        `The branch ${branchName} could not be deleted. Error: ${err.message}`
+      )
+    }
+
+    try {
+      await github.rest.repos.deleteRelease({
+        owner,
+        repo,
+        release_id: id,
+      })
+    } catch (err) {
+      return core.setFailed(
+        `The release ${id} could not be deleted. Error: ${err.message}`
+      )
+    }
```

Comment:
If all works, we should return

```suggestion
    }
    return
```

Reply:
I had asked to remove the return but you're right, we can't continue. Why are the tests succeeding with this bug? We need more tests

Reply:
The tests were not checking that the function returned early. I have added 3 more tests to confirm this.